### ISSUE TITLE
[Headless] [Logspam] SimplexNoise.Noise.Generate System.IndexOutOfRangeException Entries in Log

### ISSUE THREAD

**Issue by Enverex** [Neos-Metaverse/NeosPublic#2191]
I found that my headless log from just one session contained over 22000 lines of errors, all seemingly caused by the same issue. Here is a sample:

```
02:56:39.185    Exception updating procedural asset:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at SimplexNoise.Noise.Generate (System.Single x, System.Single y) [0x00121] in <7065cb282c784500b91cd90d70653d65>:0
  at BaseX.MathX.SimplexNoise2D (BaseX.float2& pos) [0x00000] in <7065cb282c784500b91cd90d70653d65>:0
  at FrooxEngine.SimplexTexture.UpdateTextureData (CodeX.Bitmap2D tex2D) [0x0005f] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralTextureBase.UpdateAssetData (FrooxEngine.Texture2D asset) [0x00015] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralAssetProvider`1[A].RunBackgroundAssetUpdate () [0x0002e] in <6a025c3ca62d459787cabd24f02e7564>:0
02:56:39.202    Exception updating procedural asset:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at SimplexNoise.Noise.Generate (System.Single x, System.Single y) [0x00121] in <7065cb282c784500b91cd90d70653d65>:0
  at BaseX.MathX.SimplexNoise2D (BaseX.float2& pos) [0x00000] in <7065cb282c784500b91cd90d70653d65>:0
  at FrooxEngine.SimplexTexture.UpdateTextureData (CodeX.Bitmap2D tex2D) [0x0005f] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralTextureBase.UpdateAssetData (FrooxEngine.Texture2D asset) [0x00015] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralAssetProvider`1[A].RunBackgroundAssetUpdate () [0x0002e] in <6a025c3ca62d459787cabd24f02e7564>:0
02:56:39.218    Exception updating procedural asset:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at SimplexNoise.Noise.Generate (System.Single x, System.Single y) [0x00121] in <7065cb282c784500b91cd90d70653d65>:0
  at BaseX.MathX.SimplexNoise2D (BaseX.float2& pos) [0x00000] in <7065cb282c784500b91cd90d70653d65>:0
  at FrooxEngine.SimplexTexture.UpdateTextureData (CodeX.Bitmap2D tex2D) [0x0005f] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralTextureBase.UpdateAssetData (FrooxEngine.Texture2D asset) [0x00015] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralAssetProvider`1[A].RunBackgroundAssetUpdate () [0x0002e] in <6a025c3ca62d459787cabd24f02e7564>:0
02:56:39.235    Exception updating procedural asset:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at SimplexNoise.Noise.Generate (System.Single x, System.Single y) [0x00121] in <7065cb282c784500b91cd90d70653d65>:0
  at BaseX.MathX.SimplexNoise2D (BaseX.float2& pos) [0x00000] in <7065cb282c784500b91cd90d70653d65>:0
  at FrooxEngine.SimplexTexture.UpdateTextureData (CodeX.Bitmap2D tex2D) [0x0005f] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralTextureBase.UpdateAssetData (FrooxEngine.Texture2D asset) [0x00015] in <6a025c3ca62d459787cabd24f02e7564>:0
  at FrooxEngine.ProceduralAssetProvider`1[A].RunBackgroundAssetUpdate () [0x0002e] in <6a025c3ca62d459787cabd24f02e7564>:0
```

This goes on for another 3000 times. While the error may be correct, is it possible to mitigate the issue to avoid flooding logs?

**Comment by Frooxius** [comment]
Hmm, I've ran a fuzzer on the algorithm for about 9 hours, but didn't find any issue with the algorithm. Can you send me the whole log please? It might've dumped the input state in one of the messages, so I can better tell what inputs caused it to break.

Was the headless running on Windows or Linux? There might be some difference in handling floating point values there that caused this to break.

**Comment by Frooxius** [comment]
The logspam should be fixed in 2021.5.9.179 now, when procedural assets encounter exceptions they will stop updating. They also print out more information, so please send me logs if you or anyone else runs into the procedural asset stopping to update.

Unfortunately I couldn't find the underlying cause for the procedural texture throwing this exception, so I can't address that one until we get more information.
